controllers/payload.js

```diff
@@ -0,0 +1,38 @@
+var data;
+var flag=0;
+
+const iterateObject = (obj) =>{
+    for(prop in obj){
+        console.log("checking type of value in this object ")
+        console.log(obj[prop]);
+        if(obj[prop].valueType!="string"){
+            console.log("need to be iterated");
+            iterateObject(obj[prop].value);
+        }
+        else{
+            console.log("its a string");
+            console.log(obj[prop].value);
+            var valuetobereplaced = obj[prop].value;
+            for(key in data.referenceData){
+                if(valuetobereplaced.indexOf(key) != -1){
+                    flag =1;
+                    var r = "{" + key + "}";
+                    var newstr = valuetobereplaced.replace(r,data.referenceData[key])
```

Comment:
newstr is in functional scope. can you rewrite using block scope using let, const? (for performance only, logic here is fine)
also if you used 
valuetobereplaced = valuetobereplaced.replace(r,data.referenceData[key])
then, no need for newstr variable.

Reply:
ok..I have rewritten the code using block scope